### DISCUSSION TITLE
A doubt regarding `Remove key prop` in Doc

### DISCUSSION THREAD

**Discussion by zzAlexzz** [Shopify/flash-list#569] [Q&A]
Hi team,

I have a doublt reagrding `Writing performant components - Remove key prop` (https://shopify.github.io/flash-list/docs/fundamentals/performant-components/#remove-key-prop):

From my understanding:
- `map()` mean under parent component `<></>`, there would be 4 element `<Text />`
- sicne all `<Text />` has the same component type, and `key` can help React Compiler to identify which component need to be compared with last render(otherwise it cannot be optimized)
- then React will check if all parameters are the same as last render
- if parameter value is primitive type, `a === b`, they should point to the same memory address.
- so from here, as long as `item.id` is the same string, it doesn't mater if `item` has diffrent address or not

I think we should 
- use `key` prop in rendering map items, and
- just focus the props we send to a component, and see if they are the same as last rendering/

Regards,
Alex

**Comment by davebcn87** [comment]
Could you elaborate what you propose we should do by showing some code?
